plugin/pkg/global/extensionvalidation/admission_test.go

```diff
@@ -0,0 +1,381 @@
+
+			err := admissionHandler.Validate(context.TODO(), attrs, nil)
+
+			Expect(err).To(HaveOccurred())
+		})
+
+		It("should prevent the object from being created because extension type is not registered", func() {
+			attrs := admission.NewAttributesRecord(backupEntry, nil, core.Kind("BackupEntry").WithVersion("version"), backupEntry.Namespace, backupEntry.Name, core.Resource("backupentries").WithVersion("version"), "", admission.Create, &metav1.DeleteOptions{}, false, nil)
+
+			err := admissionHandler.Validate(context.TODO(), attrs, nil)
+
+			Expect(err).To(HaveOccurred())
+		})
+	})
+
+	Context("Seed", func() {
+		var seed = &core.Seed{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "seed",
+			},
+			Spec: core.SeedSpec{
+				Provider: core.SeedProvider{
+					Type: "foo",
+				},
+				Backup: &core.SeedBackup{
+					Provider: "bar",
+				},
+			},
+		}
+
+		var (
+			kindToTypes = []struct {
+				extensionKind, extensionType string
+			}{
+				{extensionsv1alpha1.ControlPlaneResource, seed.Spec.Provider.Type},
+				{extensionsv1alpha1.BackupBucketResource, seed.Spec.Backup.Provider},
+				{extensionsv1alpha1.BackupEntryResource, seed.Spec.Backup.Provider},
+			}
+			registerAllExtensions = func() {
+				for _, registration := range kindToTypes {
+					controllerRegistration := createControllerRegistrationForKindType(registration.extensionKind, registration.extensionType)
+					gardenExternalCoreInformerFactory.Core().V1beta1().ControllerRegistrations().Informer().GetStore().Add(controllerRegistration)
+				}
+			}
+		)
+
+		It("should allow to create the object", func() {
+			registerAllExtensions()
+
+			attrs := admission.NewAttributesRecord(seed, nil, core.Kind("Seed").WithVersion("version"), seed.Namespace, seed.Name, core.Resource("seeds").WithVersion("version"), "", admission.Create, &metav1.DeleteOptions{}, false, nil)
+
+			err := admissionHandler.Validate(context.TODO(), attrs, nil)
+
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		It("should prevent the object from being created because some extension is not registered", func() {
+			for _, registration := range kindToTypes {
+				registerAllExtensions()
+
+				controllerRegistration := createControllerRegistrationForKindType(registration.extensionKind, registration.extensionType)
+				gardenExternalCoreInformerFactory.Core().V1beta1().ControllerRegistrations().Informer().GetStore().Delete(controllerRegistration)
+
+				attrs := admission.NewAttributesRecord(seed, nil, core.Kind("Seed").WithVersion("version"), seed.Namespace, seed.Name, core.Resource("seeds").WithVersion("version"), "", admission.Create, &metav1.DeleteOptions{}, false, nil)
+
+				err := admissionHandler.Validate(context.TODO(), attrs, nil)
+
+				Expect(err).To(HaveOccurred(), fmt.Sprintf("expected that extension %s is not registered", controllerRegistration.Name))
```

Comment:
Can you add something like this to ensure the validation fails because of the current `registration`?

```suggestion
				Expect(err).To(HaveOccurred(), fmt.Sprintf("expected that extension %s is not registered", controllerRegistration.Name))
				Expect(err).To(ContainSubstring(controllerRegistration.Name))
```